recipes/ginkgo/all/conanfile.py

```diff
@@ -74,13 +83,27 @@ def loose_lt_semver(v1, v2):
             min_length = min(len(lv1), len(lv2))
             return lv1[:min_length] < lv2[:min_length]
 
-        minimum_version = self._minimum_compilers_version.get(str(self.settings.compiler))
-        if minimum_version and loose_lt_semver(str(self.settings.compiler.version), minimum_version):
+        minimum_version = self._minimum_compilers_version.get(
+            str(self.settings.compiler)
+        )
+        if minimum_version and loose_lt_semver(
+            str(self.settings.compiler.version), minimum_version
+        ):
             raise ConanInvalidConfiguration(
                 f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
             )
 
-        if self.options.shared and is_msvc(self) and is_msvc_static_runtime(self):
+        is_windows_shared = self.options.shared and is_msvc(self)
+
+        if (
+            is_windows_shared
+            and self.settings.build_type == "Debug"
+            and Version(self.version) >= "1.7.0"
+        ):
+            raise ConanInvalidConfiguration(
+                "Ginkgo >= 1.7.0 cannot be built in shared debug mode on Windows"
+            )
```

Comment:
What kind of error it produces? Is it something disable by the upstream or only affecting this new version?

Reply:
```
LINK : fatal error LNK1189: library limit of 65535 objects exceeded [C:\J2\w\prod-v1\bsr@2\51653\accad\.conan\data\ginkgo\1.7.0\_\_\build\4f1710918aa542fccb5a54d7bd712e4b0750b50d\build\core\ginkgo.vcxproj]
ginkgo/1.7.0: 
```
see https://github.com/conan-io/conan-center-index/pull/21058#issuecomment-1812003537

Reply:
This is the same approach we take for MSVC in vcpkg, where we only allow static builds altogether.